backend/api/repositories/accounts.py

```
@@ -53,90 +58,136 @@ def update(account: Account, **kwargs) -> Account:
+        "FUNCTION-CALL": "accounts.registration().user-payload",
+        "payload": payload
+    })
 
-    # Create a new user
+    user_result = users.create(payload)
 
+    # Create a new store
     payload = {
-        "account_id": utils.v(account_result, "account_id"),
-        "email": data["email"],
+        "account_id": utils().get_value(data=account_result, key="account_id"),
+        "store_name": utils().get_value(data=data, key="account_name")
     }
 
-    user_result = users.create(payload)
+    current_app.logger.debug({
+        "FUNCTION-CALL": "accounts.registration().store-payload",
+        "payload": payload
+    })
+
+    store_result = stores.create(payload)
+
+    response = {**account_result, **user_result, **store_result}
 
-    response["user"] = user_result
+    if errors:
+        response = {**response, **{"errors": errors}}
 
     return response
 
 
-def create(data: dict) -> Account:
+def exists(data, errors) -> Any:
+    account_name = utils().get_value(data, "account_name", "").lower()
+
+    found = False
+    try:
+        account = accounts.get_by(name=account_name)
+        if account is not None:
+            utils().add_error(errors, "account", "Account name already exists")
+        found = True
+    except NoResultFound:
+        pass
+
+    return found
+
+
+def create(data: dict) -> Union[Account, dict, list, None]:
     """
-    Create a new account, user and store
+    Create a new account
     """
 
-    data_validation = {"email": data["email"]}
-
     # Instantiate the schema
     schema = AccountCreateSchema()
 
-    # Validate an email
-    try:
-        result = schema.load(data_validation)
-    except ValidationError as err:
-        account_errors = [
-            {"ref": ref, "message": msg}
-            for ref, msgs in err.messages.items()
-            for msg in msgs
-        ]
+    country = pycountry.countries.get(alpha_2=data.get("country").upper())
+    if country is None:
```

This valation should be on the schema level.
I would remove from here and let it fail.

---

basically if the country is not found, I am assuming Portugal by default